modin/core/dataframe/pandas/dataframe/dataframe.py

```diff
@@ -154,9 +154,11 @@ class PandasDataframe(ClassLogger):
     ----------
     partitions : np.ndarray
         A 2D NumPy array of partitions.
-    index : sequence, optional
+    index : sequence or callable, optional
         The index for the dataframe. Converted to a ``pandas.Index``.
         Is computed from partitions on demand if not specified.
+        If ``callable() -> (pandas.Index, list or None)`` type, then the
+        calculation will be delayed until `self.index` is called.
```

Comment:
```suggestion
        If ``callable() -> (pandas.Index, list or row lengths or None)`` type,
        then the calculation will be delayed until `self.index` is called.
```

Reply:
The list type is the row lengths that are stored in it.

Reply:
Corrected the suggestion. It is not clear what is the list. "list of row lengths" or just "row lengths" looks much clearer to me.